pd/src/pending_block.rs

```diff
@@ -24,6 +24,10 @@ pub struct PendingBlock {
     pub epoch: Option<Epoch>,
     /// Indicates the duration in blocks of each epoch.
     pub epoch_duration: u64,
+    /// If this is the last block of an epoch, base rates for the next epoch go here.
+    pub next_base_rate: Option<BaseRateData>,
+    /// If this is the last block of an epoch, validator rates for the next epoch go here.
+    pub next_rates: Option<Vec<RateData>>,
```

Comment:
Adding this to the `PendingBlock` means that the `State` doesn't need to compute it, xref https://github.com/penumbra-zone/penumbra/pull/321/files#r777265332